BLUESPAWN-client/src/util/configurations/RegistryKey.cpp

```diff
@@ -313,7 +328,7 @@ namespace Registry {
 
 	template<>
 	bool RegistryKey::SetValue(const std::wstring& name, LPCWSTR value, DWORD size, DWORD type) const {
-		return RegistryKey::SetRawValue(name, { PBYTE(value), wcslen(value), AllocationWrapper::STACK_ALLOC }, type);
+		return RegistryKey::SetRawValue(name, { PBYTE(value), wcslen(value) * 2 + 2, AllocationWrapper::STACK_ALLOC }, type);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5196721)